apps/shelter-web/src/app/shared/components/map/map.tsx

```diff
@@ -61,6 +66,24 @@ export function Map(props: TMap) {
     console.info(`[map] loading status: ${mapApiStatus}`);
   }, [mapApiStatus]);
 
+  const handleClick = (markerId: string | null | undefined) => {
```

Comment:
almost, but we prefer for `Map` to know nothing about shelters or even modals.
so better to define it outside (home.tsx ok for now), and here we'd just update the AdvancedMarker:

```javascript
      {markers.map((marker) => (
        <AdvancedMarker
          key={marker.id}
          onClick={marker.onClick}
          ...
```